README.md

```diff
@@ -230,107 +230,137 @@ The `ping()` function is the only exception to this rule. It does not return hea
 
 ### Request Plugins
 
-This library can be used with one (or more) of the following `request` plugins:
+The library is easily extendable via the use of plugins. They provide the ability to intercept a request:
+1. Before the request is submitted to the server.
+2. After the response headers are received.
+3. If the underlying HTTP client emits an `error` event.
+
+Plugins can be used to modify an outgoing request, edit an incoming response or even retry a request entirely.
+
+#### Plugin Configuration
+
+The following global configurations can be defined to affect the behavior of all plugins:
+-	`maxAttempt` - The maximum number of times the request will be attempted _(default: 3)_.
+-	`retryInitialDelayMsecs` - The initial retry delay in milliseconds _(default: 500)_.
+-	`retryDelayMultiplier` - The multiplication factor used for increasing the timeout after each subsequent attempt _(default: 2)_.
+
+Note that `retryInitialDelayMsecs` and `retryDelayMultiplier` can be overriden by plugin specific configuration. This is achieved by passing the plugin (and its associated configuration) as an object.
```

Comment:
I'm planning to combine all the retry plugins, i.e.

```
plugin: [ { retry: { retryDelayMultiplier: 4, retryStatusCodes: [ 429, 500, 501 ], retryError: true } } ]
```

These settings can then be removed from the global configuration scope and be plugin specific.